packages/playwright-core/src/server/fetch.ts

```diff
@@ -515,11 +514,21 @@ export abstract class APIRequestContext extends SdkObject {
             }),
         );
 
+        // when using socks proxy, having the socket means the connection got established
+        if (agent instanceof SocksProxyAgent)
```

Comment:
this should be also the case for HttpsProxyAgent? So maybe we should check for just agent thruthyness there?

Reply:
HTTPSProxyAgent emits its own separate event that's more precise, so we use that: https://github.com/microsoft/playwright/pull/33003/files#diff-dc06386362e7da98cd82627a6b4f337ae01799a53cac4f208089b2b89c4dc832R527-R529